packages/ra-core/src/types.ts

```diff
@@ -69,11 +69,11 @@ export type AuthProvider = {
     handleCallback?: (
         params?: QueryFunctionContext
     ) => Promise<AuthRedirectResult | void | any>;
-    canAccess?: (
+    canAccess?: <RecordType extends RaRecord | Omit<RaRecord, 'id'> = RaRecord>(
```

Comment:
I wouldn't use RaRecord here, as it's possible to use an empty object.

Reply:
`Omit<RaRecord, 'id'>` is basically any object